packages/story-editor/src/app/canvas/useCanvasBoundingBox.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useCallback, useRef, useContextSelector } from '@web-stories-wp/react';
+/**
+ * Internal dependencies
+ */
+import Context from './context';
+import { RESIZE_OBSERVATION_KEY } from './constants';
+
+/**
+ * Returns a ref that when applied to an element, stores the elements
+ * bounding box info. can retieve the relevant bounding box info like so:
+ * ```js
+ * const boundingBox = useCanvasBoundingBox(CANVAS_BOUNDING_BOX_IDS[<some_bounding_box_id>])
+ * ```
+ *
+ * @param {string} boundingBoxId a value from CANVAS_BOUNDING_BOX_IDS
+ * @return {Function} a callback ref
+ */
+export function useCanvasBoundingBoxRef(boundingBoxId) {
+  const resizeObserver = useContextSelector(
+    Context,
+    (v) => v.state.resizeObserver
+  );
+  const ref = useRef(null);
+  return useCallback(
+    (node) => {
+      if (ref.current) {
+        resizeObserver.unobserve?.(ref.current);
+      }
+
+      if (node) {
+        node.dataset[RESIZE_OBSERVATION_KEY] = boundingBoxId;
+        resizeObserver.observe(node);
+      }
+
+      ref.current = node;
+    },
+    [resizeObserver, boundingBoxId]
+  );
+}
+
+/**
+ * Returns the bounding box associated with the relative CANVAS_BOUNDING_BOX_IDS
+ *
+ * @param {string} boundingBoxId a value from CANVAS_BOUNDING_BOX_IDS
+ * @return {Object} bounding box associated with `boundingBoxId`
+ */
+export function useCanvasBoundingBox(boundingBoxId) {
+  return useContextSelector(
+    Context,
+    (v) => v.state.boundingBoxes[boundingBoxId]
```

Comment:
looks like this is throwing off some unit tests

https://github.com/google/web-stories-wp/runs/4652759966?check_suite_focus=true#step:6:6874
<img width="647" alt="Screen Shot 2022-01-03 at 12 57 51 PM" src="https://user-images.githubusercontent.com/10720454/147974606-a0ec8aee-43c8-4627-9300-24d28d9ab58a.png">